packages/core/src/types.ts

```diff
@@ -38,12 +38,17 @@ export interface ParticipantClickEvent {
   track?: TrackPublication;
 }
 
-export type TrackObserverOptions =
-  | { source: Track.Source; name?: undefined }
-  | { source?: undefined; name: string };
+export type TrackSource<T extends Track.Source> = RequireAtLeastOne<
+  { source: T; name: string },
+  'name' | 'source'
+>;
+
+export type TrackIdentifier<T extends Track.Source = Track.Source> =
```

Comment:
nit: would be helpful to include a bit of comments showing examples of what would qualify. Since I imagine users would click into `TrackIdentifier` and try to understand what it means.